src/Diagnostics.DataProviders/DataProviders.cs

```diff
@@ -18,27 +20,85 @@ public class DataProviders
         public Func<MdmDataSource, IMdmDataProvider> Mdm;
         public Func<GenericMdmDataProviderConfiguration, IMdmDataProvider> MdmGeneric;
 
+        private readonly Dictionary<object, IMetadataProvider> _dataProvidersCache = new Dictionary<object, IMetadataProvider>();
```

Comment:
This Cache is per Request? 
We create Data Providers object for each request.

Reply:
Yes.